packages/visualizations/src/stores/createDeepEqual.ts

```diff
@@ -0,0 +1,29 @@
+import { writable, get } from 'svelte/store';
+
+/**
+ * Creates a Svelte writable store that compares values deeply before updating.
+ *
+ * @param initialValue - The initial value for the store.
+ * @returns An object containing the subscribe and set methods.
+ */
+const createDeepEqual = <V>(initialValue: V | undefined) => {
+    const value = writable<V | undefined>(initialValue);
+
+    return {
+        /**
+         * Subscribes to changes in the store's value.
+         */
+        subscribe: value.subscribe,
+        /**
+         * Sets the new value for the store if it differs from the current value
+         * by performing a deep comparison.
+         */
+        set: (newValue: V | undefined) => {
+            if (JSON.stringify(newValue) !== JSON.stringify(get(value))) {
+                value.set(newValue);
+            }
+        },
+    };
+};
+
+export default createDeepEqual;
```

Comment:
```suggestion
import { writable, get } from 'svelte/store';
import { isEqual } from 'lodash';
/**
 * Creates a Svelte writable store that compares values deeply before updating.
 *
 * @param initialValue - The initial value for the store.
 * @returns An object containing the subscribe and set methods.
 */
const createDeepEqual = <V>(initialValue: V | undefined) => {
    const { subscribe, set} = writable<V | undefined>(initialValue);

    return {
        /**
         * Subscribes to changes in the store's value.
         */
        subscribe,
        set,
        /**
         * Sets the new value for the store if it differs from the current value
         * by performing a deep comparison.
         */
        update: (value: V, newValue?: V) => {
            if (isEqual(value, newValue)) {
                return newValue;
            }
            return value;
        },
    };
};

export default createDeepEqual;
```
I think using update is a bit more efficient, since it avoids reading the whole store in `set`. Also, in the long term `isEqual` is both more efficient and also covers objects, Sets and Maps with different ordering etc.

Reply:
Thx, I used store.update() to get the current store value https://github.com/opendatasoft/ods-dataviz-sdk/pull/194/commits/f8f0f37163ab9dfb0c82bee2f60f77ccf95d327b